app/models/llanfair.rb

```diff
@@ -0,0 +1,113 @@
+module Llanfair
+  class Run < Run
+    def self.sti_name
+      :llanfair
+    end
+  end
+
+  class Split < Split
+  end
+
+  class Parser < BabelBridge::Parser
+    def parse(character_array)
+      # Convert the array of character from string form to an actual array
+      # Then pack the characters back to the original bytes
+      file = RunFile.pack_binary(character_array)
+      run = {}
+      run[:time] = 0
+      # A port of Nitrofski's import llanfair files function for wsplit
+      file = StringIO.new(file)
+      file.seek(197, 1)
+      goal_len = file.read(2).unpack("n")[0]
+      run[:goal] = file.read(goal_len)
+
+      file.seek(1, 1)
+      title_len = file.read(2).unpack("n")[0]
+      run[:title] = file.read(title_len)
```

Comment:
Don't bother setting a `run[:title]`, we just call it `run[:name]` no matter what the program likes to call it, so that runs from different programs have mostly the same structure after parsing.